app/src/main/java/com/li/musicmp3/ui/main/SongAdapter.kt

```diff
@@ -0,0 +1,49 @@
+package com.li.musicmp3.ui.main
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.li.musicmp3.R
+import com.li.musicmp3.data.model.Song
+import kotlinx.android.synthetic.main.item_song.view.*
+
+class SongAdapter(
+    val handleClick: (position: Int) -> Unit
+) : RecyclerView.Adapter<SongAdapter.ViewHolder>() {
+
+    private var songs = mutableListOf<Song>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_song, parent, false)
+        return ViewHolder(view)
+    }
+
+    override fun getItemCount(): Int = songs.size
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.onBind(songs[position])
+    }
+
+    fun updateData(songs: List<Song>) {
+        this.songs = songs.toMutableList()
+        notifyDataSetChanged()
+    }
+
+    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
không dùng inner tránh memory leak 
```suggestion
   class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```